parlai/core/dict.py

```diff
@@ -827,7 +824,7 @@ def __str__(self):
         """
         return str(self.freq)
 
-    def set_tokenization_mode(self, mode: TokenizationMode):
+    def set_tokenization_mode(self, mode: TokenizationMode, resample_bpe: bool = False):
```

Comment:
This may not be the best way to abstract ---  it feels a bit weird to put a bpe related thing in the dict/agent level but I'm not sure if there's a better way.

Reply:
Yeah I didn't like the abstraction leak either. That's why I was trying to signal it on the TokenizationMode